lib/datadog/core/configuration/agent_settings_resolver.rb

```diff
@@ -201,8 +201,12 @@ def timeout_seconds
         # In transport_options, we try to invoke the transport_options proc and get its configuration. In case that
         # doesn't work, we include the proc directly in the agent settings result.
         def deprecated_for_removal_transport_configuration_proc
-          if settings.tracing.transport_options.is_a?(Proc) && transport_options.adapter.nil?
-            settings.tracing.transport_options
+          transport_options_settings if transport_options_settings.is_a?(Proc) && transport_options.adapter.nil?
+        end
+
+        def transport_options_settings
+          @transport_options_settings ||= begin
+            settings.tracing.transport_options if settings.respond_to?(:tracing) && settings.tracing
           end
```

Comment:
This is a bit of a regression in the making... Right now, we always load `tracing/configuration/settings` from `core/configuration/settings` so there is no way for `settings.respond_to?(:tracing)` to ever be `false`.

So currently, this code is equivalent to what was before.

BUT, I suggest being careful with next steps. [I had suggested in a previous comment](https://github.com/DataDog/dd-trace-rb/pull/3049#issuecomment-1683553855), we're still  [documenting the `transport_options`](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/#configuring-the-transport-layer) as a valid way of configuring dd-trace-rb -- that includes all products, so we need to be careful with the way we plan to make `settings.respond_to?(:tracing)` start returning `false` to not impact existing customers.

(This assuming we plan to touch that before 2.0; if we move things around for 2.0 that seems fine)

Reply:
We don't plan on removing `transport_options` prior to 2.0. But we do plan on refactoring the Settings class and removing hard dependencies on tracing (I haven't fully researched the idea yet, it's on the list). This should have no effect on users prior to 2.0. 

I don't think this is a dangerous change given then fact that our tests for the `transport_options` proc still expect `settings.tracing` to exist and the rest of resolver code treats `transport_options` as optional.

Reply:
Yeah, agreed!